gateway/handler_unixfs_dir.go

```diff
@@ -59,42 +60,60 @@ func (i *handler) serveDirectory(ctx context.Context, w http.ResponseWriter, r *
 	}
 
 	// Check if directory has index.html, if so, serveFile
-	idxPath := ipath.Join(contentPath, "index.html")
-	imIndexPath, err := NewImmutablePath(ipath.Join(resolvedPath, "index.html"))
+	appendIndexHtml := func(p ipath.Path) ipath.Path {
+		basePath := p.String()
+		if basePath[len(basePath)-1] == '/' {
+			return ipath.New(basePath + "index.html")
+		}
+		return ipath.New(basePath + "/index.html")
+	}
+
+	idxPath := appendIndexHtml(contentPath)
+	imIndexPath, err := NewImmutablePath(appendIndexHtml(resolvedPath))
```

Comment:
@lidel @hacdias how's this for our `//index.html` issues? Kind of gross, but seems like it'd do the job until we have something like #334 to make this less painful.

Reply:
sgtm. we could would flip it to append index.html only once:

```suggestion
	appendIndexHtml := func(p ipath.Path) ipath.Path {
		basePath := p.String()
		if basePath[len(basePath)-1] != '/' {
			basePath += "/"
		}
		return ipath.New(basePath + "index.html")
	}

	idxPath := appendIndexHtml(contentPath)
	imIndexPath, err := NewImmutablePath(appendIndexHtml(resolvedPath))
```